lib/diplomat/entity.ex

```diff
@@ -72,30 +70,26 @@ defmodule Diplomat.Entity do
       properties: pb_properties
     }
   end
-
-  @doc false
-  def proto(%{} = properties) do
-    proto(%Entity{key: nil, properties: properties})
+  def proto(properties) when is_map(properties) do
+    properties
+    |> new()
+    |> proto()
   end
 
-  @spec from_proto(Diplomat.Proto.Entity.t) :: t
+  @spec from_proto(PbEntity.t) :: t
   @doc "Create a `Diplomat.Entity` from a `Diplomat.Proto.Entity`"
+  def from_proto(%PbEntity{key: nil, properties: pb_properties}),
```

Comment:
A small refactoring of `from_proto` to avoid having to lean on `Key.proto` in the nil key case.